src/clr-angular/forms/styles/_input.clarity.scss

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+// This software is released under MIT license.
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@include exports('forms.control.input') {
+    .clr-input-wrapper {
+        white-space: nowrap;
+        height: $clr-forms-baseline * 4;
+        
+        .clr-input {
+            @include custom-inputs-reset($clr-forms-baseline * 4);
+            @include form-fields-appearance(2.5rem, $clr-forms-border-color);
+            padding: 0 $clr-forms-baseline;
+            max-width: calc(100% - #{$clr-forms-icon-size});
+            margin-right: $clr-forms-icon-size;
+            max-height: $clr-forms-baseline * 4;
+
+            @include clr-getTypePropertiesForDomElement(form_label, (font-size));
+            
+            &[readonly] {
+                border: none;
+            }
+            
+            &:not([readonly]) {
+                @include input-border-bottom-animation($clr-forms-focused-color);
+            }
+        }
+    }
+
+    .clr-control-container.clr-error .clr-input-wrapper .clr-input {
```

Comment:
If we end up removing the above nesting, we could just use:

`.clr-control-container.clr-error .clr-input`

instead of `.clr-control-container.clr-error .clr-input-wrapper .clr-input`

Reply:
Done.